src/ngx_http_lua_common.h

```diff
@@ -176,6 +178,9 @@ struct ngx_http_lua_main_conf_s {
 
     ngx_array_t         *preload_hooks; /* of ngx_http_lua_preload_hook_t */
 
+    ngx_array_t         *delay_list; /* of void* */
+    unsigned             delay_list_inited;
```

Comment:
@rainingmaster Using an array to count registrations is also very strange to me. It just does not make any sense. If you really want to count registrations, we should just maintain a single counter instead of an array without any useful data in its elements. Am I overlooking anything here?